lib/src/rad/script.dart

```diff
@@ -23,81 +19,238 @@ class RadType{
-  RadonArray keys() {return RadonArray(concat(script, OP.MAP_KEYS));}
-  RadonArray valuesAsArray() {return RadonArray(concat(script, OP.MAP_VALUES_AS_ARRAY));}
+
+const List emptyList = [];
+
+class RadonMap extends RadType {
+  RadonMap([List script = emptyList]) : super(script);
+  RadonArray getArray(String key) =>
+      RadonArray(concat(script, [OP.MAP_GET_ARRAY, key]));
+  RadonBoolean getBoolean(String key) =>
+      RadonBoolean(concat(script, [OP.MAP_GET_BOOLEAN, key]));
+  RadonBytes getBytes(String key) =>
+      RadonBytes(concat(script, [OP.MAP_GET_BYTES, key]));
+  RadonFloat getFloat(String key) =>
+      RadonFloat(concat(script, [OP.MAP_GET_FLOAT, key]));
+  RadonInteger getInteger(String key) =>
+      RadonInteger(concat(script, [OP.MAP_GET_INTEGER, key]));
+  RadonMap getMap(String key) =>
+      RadonMap(concat(script, [OP.MAP_GET_MAP, key]));
+  RadonString getString(String key) =>
+      RadonString(concat(script, [OP.MAP_GET_STRING, key]));
+  RadonArray keys() => RadonArray(concat(script, OP.MAP_KEYS));
+  RadonArray valuesAsArray() =>
+      RadonArray(concat(script, OP.MAP_VALUES_AS_ARRAY));
 }
-class RadonString extends RadType{
+
+class RadonString extends RadType {
   RadonString([List? script]) : super((script == null) ? [] : script);
-  RadonBoolean asBoolean() {return RadonBoolean(concat(script, OP.STRING_AS_BOOLEAN));}
-  RadonFloat asFloat() {return RadonFloat(concat(script, OP.STRING_AS_FLOAT));}
-  RadonInteger asInteger() {return RadonInteger(concat(script, OP.STRING_AS_INTEGER));}
-  RadonInteger length() {return RadonInteger(concat(script, OP.STRING_LENGTH));}
-  RadonBoolean match(String key) {return RadonBoolean(concat(script, [OP.STRING_MATCH, key]));}
-  RadonArray parseJSONArray() {return RadonArray(concat(script, OP.STRING_PARSE_JSON_ARRAY));}
-  RadonMap parseJSONMap() {return RadonMap(concat([], OP.STRING_PARSE_JSON_MAP));}
-
-  RadonString toLowerCase() {return RadonString(concat(script, OP.STRING_TO_LOWER_CASE));}
-  RadonString toUpperCase() {return RadonString(concat(script, OP.STRING_TO_UPPER_CASE));}
+  RadonBoolean asBoolean() =>
+      RadonBoolean(concat(script, OP.STRING_AS_BOOLEAN));
+  RadonFloat asFloat() => RadonFloat(concat(script, OP.STRING_AS_FLOAT));
+  RadonInteger asInteger() =>
+      RadonInteger(concat(script, OP.STRING_AS_INTEGER));
+  RadonInteger length() => RadonInteger(concat(script, OP.STRING_LENGTH));
+  RadonBoolean match(Map<String, dynamic> key, bool state) =>
+      RadonBoolean(concat(script, [OP.STRING_MATCH, key, state]));
+  RadonArray parseJSONArray() =>
+      RadonArray(concat(script, OP.STRING_PARSE_JSON_ARRAY));
+  RadonMap parseJSONMap() => RadonMap(concat([], OP.STRING_PARSE_JSON_MAP));
+  RadonString toLowerCase() =>
+      RadonString(concat(script, OP.STRING_TO_LOWER_CASE));
+  RadonString toUpperCase() =>
+      RadonString(concat(script, OP.STRING_TO_UPPER_CASE));
+}
+
+class Request {}
```

Comment:
it is not being used and we can remove it. 
initially i guess it was there since i try to mimic the witnet-requests-js library but ended up implementing `Witnet.Request()` on line #183